src/Elastic.Apm/Metrics/MetricsProvider/CgroupMetricsProvider.cs

```diff
@@ -82,6 +90,14 @@ internal class CgroupMetricsProvider : IMetricsProvider
 
 		private CgroupFiles FindCGroupFiles(string procSelfCGroup, string mountInfo)
 		{
+			if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && !_ignoreOs)
+			{
+				_logger.Trace()
+					?.Log("{MetricsProviderName} detected a non Linux OS, therefore"
+						+ " Cgroup metrics will not be reported", nameof(CgroupMetricsProvider));
+				return null;
+			}

```

Comment:
This looks like it could moved to the start of the internal ctor, and return early if the condition is matched.

Reply:
Makes sense - done.